rust/arrow/src/compute/util.rs

```diff
@@ -71,6 +71,47 @@ pub(super) fn combine_option_bitmap(
     }
 }
 
+/// Compares the null bitmaps of two arrays using a bitwise `or` operation.
+///
+/// This function is useful when implementing operations on higher level arrays.
+pub(super) fn compare_option_bitmap(
+    left_data: &ArrayDataRef,
+    right_data: &ArrayDataRef,
+    len_in_bits: usize,
+) -> Result<Option<Buffer>> {
+    let left_offset_in_bits = left_data.offset();
+    let right_offset_in_bits = right_data.offset();
+
+    let left = left_data.null_buffer();
+    let right = right_data.null_buffer();
+
+    if (left.is_some() && left_offset_in_bits % 8 != 0)
```

Comment:
Out of curiosity: is this true in general, or is for these (this and above) implementation, that uses this assumption?

Reply:
I'm trying to solve this problem in #8262 

The issue is that `buffer.slice` and `buffer_bin_or/and` currently work with byte offsets, but for boolean arrays and bitmaps the offset can start in the middle of a byte.